motr/io_nw_xfer.c

```diff
@@ -1169,11 +1169,12 @@ static int nw_xfer_tioreq_get(struct nw_xfer_request *xfer,
 		rc = target_ioreq_init(ti, xfer, fid, ta_obj, session, size);
 		if (rc == 0) {
 			tioreqht_htable_add(&xfer->nxr_tioreqs_hash, ti);
-			M0_LOG(M0_INFO, "New target_ioreq added for "FID_F,
-			       FID_P(fid));
-		}
-		else
+			M0_LOG(M0_INFO, "New target_ioreq %p added for "FID_F,
+			                ti, FID_P(fid));
+		} else {
 			m0_free(ti);
+			return M0_ERR_INFO(rc, "target_ioreq_init() failed");
```

Comment:
It is not clear how this change is fixing the panic, as the error vlaue in `rc` returns from line 1186 as well. Can you elaborate more about it?

Reply:
@madhavemuri, see the few lines just before the return - https://github.com/Seagate/cortx-motr/blob/b64b1bba78571693a3123b462402736719632b1d/motr/io_nw_xfer.c#L1179-L1181

Reply:
You meant,  rc = dgmode_rwvec_alloc_init() is being called even if target_ioreq_init() fails, right ?

Reply:
Got it, ti should not be referenced at line:1179.

Reply:
Yes, and `dgmode_rwvec_alloc_init(ti)` should not be called with it.